streaming/src/main/scala/org/apache/spark/streaming/util/TimeoutUtils.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.streaming.util
+
+
+private[streaming] object TimeoutUtils {
+  /**
+   * Waiting function with a timeout.
+   * Returns true if done, false for timeout
+   */
+  def waitUntilDone(timeout:Long, done:() => Boolean): Boolean = {
```

Comment:
I mentioned in another comment, we can either use the `Clock` trait's `waitForTime`. Or if we want to implement something more generic like this (say `waitForCondition`), then it might better to implement in a proper class like `o.a.s.streaming.utils.ConditionWaiter` rather than something like `TimeoutUtils` which probably will have only this one method ever.